src/xtgeo/surface/regular_surface.py

```diff
@@ -2869,13 +2769,7 @@ def _ensure_correct_values(
         else:
             raise ValueError("Input values are in invalid format: {}".format(values))
 
-        if self._values is not None:
-
-            if self._ilines is None:
-                self._ilines = np.array(range(1, self._ncol + 1), dtype=np.int32)
```

Comment:
An example of why this refactor would be advantageous is that by simplifying the construction we get more of an overview of where things are actually set. Like these two lines setting `ilines` and `xlines`, they are in `__init__` and also duplicated in several (but not all) of the `import_from_*` functions.